lblod/app-digitaal-loket
# [ Maintenance ] Consolidating worship-service-sensitive delta producer

# Description
DL-5588

This PR extends the configuration of the maintainer and job-initiator for worship-services-sensitive deltas. It moves in config.json files environment variables of the docker-compose related to the services. It also removes the custom rule file to include it in the maintainer rule instead.

# Type of change


# Related services

- https://github.com/lblod/delta-producer-publication-graph-maintainer
- https://github.com/lblod/delta-producer-background-jobs-initiator
- https://github.com/lblod/delta-consumer

# How to test 

1. Add your `key` and `value` in your `app-digitaal-loket/config/delta-producer/publication-graph-maintainer/config.json` file

2. Launch Loket and OP (organisation-portal)
3. Check the logs of delta-publication-graph-maintainer if you have deltas already you wont need to create a new person and attach a worship position or mandate.
4. Depending of your delta-consumer service you'll need to set few variables 
```yml
  worship-services-sensitive-consumer:
    environment:
      DCR_DISABLE_INITIAL_SYNC: 'true'
      DCR_WAIT_FOR_INITIAL_SYNC: 'false'
      DCR_SECRET_KEY: "yourKey"
      DCR_SYNC_BASE_URL: 'http://endpoint-of-your-choice:90'
      DCR_SYNC_LOGIN_ENDPOINT: 'http://endpoint-of-your-choice:90/sync/worship-services-sensitive/login'
      DCR_START_FROM_DELTA_TIMESTAMP: "2023-11-09"
  file:
    environment:
      DISABLE_AUTOMATIC_SYNC: "true"
      DISABLE_INITIAL_SYNC: "true"
      FILES_ENDPOINT_BASE_URL: "http://endpoint-of-your-choice:90" # The endpoint of your choice (see later what to choose)
      SYNC_LOGIN_ENDPOINT: "http://endpoint-of-your-choice:90/sync/worship-services-sensitive/login"
      SECRET_KEY: "youKey"
```
5. Then log the worship-services-sensitive-consumer, it should consume the delta file created from loket.

# What to check

- N/A

# Links to other PR's

- N/A

# Notes
